test/tools/d_do_test.d

```diff
@@ -576,8 +597,9 @@ int tryMain(string[] args)
         case "compilable":              testArgs.mode = TestMode.COMPILE;      break;
         case "fail_compilation":        testArgs.mode = TestMode.FAIL_COMPILE; break;
         case "runnable":                testArgs.mode = TestMode.RUN;          break;
+        case "dshell":                  testArgs.mode = TestMode.RUN;          break;
```

Comment:
The dshell tests don't use or need any of the code below. Like the bash tests, we escape early into our own function because the logic to run them is completely different.  As you can see form the code, we do the same thing with the bash tests.

If we used a testMode instead of escaping early like this,, I'd have to disable alot of the code below behind something like `if (testMode != TestMode.dshell)` as it does things like tries to read test parameters from the input file.  And some test parameters aren't valid with dshell like DFLAGS, because if different flags are used to build the dshell test, then it wont' link to dshell_prebuilt anymore.  And there's no need to compile the dshell tests with different DFLAGS because, the point isn't to test the dshell tests, it's just to run them.

There would be some tests parameters that would make sense, like DISABLED_PLATFORMS, but that can already be done inside the test like the current bash tests do., i.e. `if (OS == "...")`.

Reply:
Oh wait, I think you just mean to add an enum value and assign it here....nevermind.  I've gone ahead and done that.